ledger/nomos-ledger/src/mantle/leader.rs

```diff
@@ -0,0 +1,122 @@
+use crate::Balance;
+use cryptarchia_engine::Epoch;
+use mmr::MerkleMountainRange;
+use nomos_core::mantle::{
+    ops::leader_claim::{LeaderClaimOp, RewardsRoot, VoucherNullifier, VoucherCm},
+    Value,
+};
+use std::cmp::Ordering;
+
+
+#[cfg_attr(feature = "serde", derive(serde::Serialize, serde::Deserialize))]
+#[derive(Clone, Debug, PartialEq, Eq)]
+pub struct LeaderState {
+    // current epoch
+    epoch: Epoch,
+    // vouchers that can be claimed in this epoch
+    // this is updated once at the start of each epoch from the root of
+    // the vouchers merkle tree
+    claimable_vouchers_root: RewardsRoot,
+    n_claimable_vouchers: u64,
```

Comment:
What do you think of storing the snapshot of the MMR here rather than these derived fields? The MMR is pretty lightweight, and then you can compute these values when you need them rather than storing the derived field.

```suggestion
    claimable_vouchers: MerkleMountainRange<VoucherCm, nomos_core::crypto::Hasher>,
```

Reply:
Yeah that was my initial idea, not sure what happened

Reply:
This way is fine as well

Reply:
Just leaving the rationale here for the future:

you would need to compute the frontier root each block instead of maintaining one more field. I'd say both are fine although I have a slight preference for the current method since it's still quite easy to understand.